packages/ListBox/src/hooks/hooks.js

```diff
@@ -7,17 +8,22 @@ function waitForPopoverAnimation(func) {
   setTimeout(func, 300);
 }
 
-export function useChildrenLengthChange(children) {
+export function useChildrenChange(children) {
   const [state, dispatch] = useListBox();
   React.useEffect(() => {
-    if (Object.keys(state.options).length !== React.Children.count(children)) {
-      const options = getDataOptions(children);
+    const options = getDataOptions(children);
 
-      dispatch({
-        type: useListBox.types.updateOptions,
-        payload: options,
-      });
+    if (Object.keys(state.options).length === Object.keys(options).length) {
```

Comment:
ohhh nice, way better approach allison 💯 ... so now we don't need to do the approach that we did for `Load more` when we have to add an `isHidden` on the option. Because now it is tracking the probs as well.

Reply:
It shouldn't impact too much the performance because it will be run only when the children change.